docs/reference/remap2/expressions/logical.cue

```diff
@@ -37,16 +37,23 @@ remap2: expressions: logical: {
 		{
 			title: "AND"
 			source: #"""
-				true && true
+				true && "foo"
```

Comment:
While we do support non-boolean types in boolean-or operations, we don't actually support it for boolean-and.

We can add it, we just haven't done so yet (and there also might be less of a use-case for this).

Reply:
To clarify:

```
false && "foo" # error
false && false # false
false || "foo" # "foo"
true || "foo" # true
!("foo") # error
!true # false
```